src/Constructorio_NET.Tests/client/modules/BrowseTests.cs

```diff
@@ -75,6 +74,48 @@ public async Task GetBrowseResultsWithFilters()
             Assert.IsNotNull(res.ResultId, "ResultId should exist");
         }
 
+        [Test]
+        public async Task GetBrowseResultsWithPreFilterExpression()
+        {
+            JObject preFilterExpression = JObject.Parse(@"{
+                or: [
+                    {
+                    and:
+                        [
+                        { name: 'group_id', value: 'BrandXY' },
+                        { name: 'Color', value: 'red' },
+                    ],
+                    },
+                    {
+                    and:
+                        [
+                        { name: 'Color', value: 'blue' },
+                        { name: 'Brand', value: 'XYZ' },
+                    ],
+                    },
+                ],
+            }");
```

Comment:
hmm, I have some ideas on this but I'm not sure which is the better approach. On one hand, this is pretty clear cut as long as the consumer formats it right, on the other, it's pretty against the strict-type style of C# :/ Could we pair on this for a bit to exchange ideas?